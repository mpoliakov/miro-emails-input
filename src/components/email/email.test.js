import Email from './email';

describe('Testing Email component...', () => {
  it('Email component snapshot', () => {
    const emailComponent = new Email('john@miro.com', null, 'emails-input__email');
    expect(emailComponent.getElement()).toMatchSnapshot();
  });

  it('Click on delete button removes element and calls delete handler', () => {
    const handleDelete = jest.fn();
    const emailComponent = new Email('john@miro.com', handleDelete, 'emails-input__email');
    emailComponent.getElement().querySelector('.email__btn-delete').click();
    expect(handleDelete).toBeCalledTimes(1);
    expect(handleDelete).toBeCalledWith('john@miro.com');
    expect(emailComponent._element).toEqual(null);
  });
});
