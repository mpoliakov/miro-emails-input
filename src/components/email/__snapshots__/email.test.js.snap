// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Email component... Email component snapshot 1`] = `
<div
  class="emails-input__email email"
>
  john@miro.com
        
  <button
    class="email__btn-delete"
    title="Delete email"
  >
    
          
    <svg
      fill="none"
      height="8"
      viewBox="0 0 8 8"
      width="8"
      xmlns="http://www.w3.org/2000/svg"
    >
      
            
      <path
        clip-rule="evenodd"
        d="M8 0.8L7.2 0L4 3.2L0.8 0L0 0.8L3.2 4L0 7.2L0.8 8L4 4.8L7.2 8L8 7.2L4.8 4L8 0.8Z"
        fill="#050038"
        fill-rule="evenodd"
      />
      
          
    </svg>
    
        
  </button>
  
    
</div>
`;
